src/syntax/FStar.Syntax.Embeddings.fs

```diff
@@ -708,6 +711,142 @@ let e_range =
         Range.string_of_range
         (ET_app (PC.range_lid |> Ident.string_of_lid, []))
 
+let e_vconfig =
```

Comment:
Is it really worth embedding and unembedding vconfig back and forth?

Why not expose a vconfig to metaprograms as an abstract type, with operations to get and set specific options working by manipulating the native object?

Reply:
This may also have the benefit of providing an interface to vconfigs for use in NBE

Reply:
The main advantage in my mind is that one can then use record syntax (projection, override, etc) and create one from scratch. But yes, the embeddings are a pain to write. Some times I considered having a small language to autogenerate these embeddings, maybe it's time to bite that bullet.